cmd/translate/main.go

```diff
@@ -9,29 +9,46 @@ import (
 	"os"
 	"time"
 
+	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
+	"github.com/soheilhy/cmux"
 	"github.com/spf13/cobra"
 	"github.com/spf13/viper"
 	pb "go.expect.digital/translate/pkg/server/translate/v1"
+	"go.expect.digital/translate/pkg/tracer"
 	"go.expect.digital/translate/pkg/translate"
-
-	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
-	"github.com/soheilhy/cmux"
+	"go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc"
+	tracesdk "go.opentelemetry.io/otel/sdk/trace"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/credentials/insecure"
 )
 
-var cfgFile string
+var (
+	cfgFile string
+	tp      *tracesdk.TracerProvider
+)
 
 // rootCmd represents the base command when called without any subcommands.
 var rootCmd = &cobra.Command{
 	Use:   "translate",
 	Short: "Enables translation for Cloud-native systems",
 	Long:  `Enables translation for Cloud-native systems`,
 	Run: func(cmd *cobra.Command, args []string) {
-		grpcSever := grpc.NewServer()
+		var err error
+
+		tp, err = tracer.TracerProvider("http://localhost:14268/api/traces", "translate")
+		if err != nil {
+			log.Panic(err)
+		}
+
+		grpcSever := grpc.NewServer(
+			grpc.UnaryInterceptor(otelgrpc.UnaryServerInterceptor()),
+			grpc.StreamInterceptor(otelgrpc.StreamServerInterceptor()),
+		)
+
 		mux := runtime.NewServeMux()
 		pb.RegisterTranslateServiceServer(grpcSever, &translate.TranslateServiceServer{})
-		err := pb.RegisterTranslateServiceHandlerFromEndpoint(
+
+		err = pb.RegisterTranslateServiceHandlerFromEndpoint(
```

Comment:
check error

Reply:
Is already checked in
https://github.com/translate-agent/translate/blob/eb1cd85852fc834c86b8a44886c47fc618dae38e/cmd/translate/main.go#L56-L58